src/pymor/reductors/neural_network.py

```diff
@@ -441,20 +518,22 @@ class NeuralNetworkStatefreeOutputReductor(NeuralNetworkReductor):
         See :class:`~pymor.reductors.neural_network.NeuralNetworkReductor`.
     """
 
-    def __init__(self, fom=None, training_set=None, validation_set=None, validation_ratio=0.1,
-                 validation_loss=None, scale_inputs=True, scale_outputs=False):
+    def __init__(self, fom=None, training_set=None, validation_set=None, training_snapshots=None,
+                 validation_snapshots=None, validation_ratio=0.1, T=None, nt=1, validation_loss=None,
+                 scale_inputs=True, scale_outputs=False):
         assert 0 < validation_ratio < 1 or validation_set
 
         self.scaling_parameters = {'min_inputs': None, 'max_inputs': None,
                                    'min_targets': None, 'max_targets': None}
 
+        super().__init__(fom=fom, training_set=training_set, validation_set=validation_set,
+                         training_snapshots=training_snapshots, validation_snapshots=validation_snapshots,
+                         validation_ratio=validation_ratio, T=T, nt=nt, scale_inputs=scale_inputs,
+                         scale_outputs=scale_outputs)
+        self.validation_loss = validation_loss
         if not fom:
-            assert training_set is not None
-            assert len(training_set) > 0
-            self.parameters_dim = training_set[0][0].parameters.dim
-            self.dim_output = len(training_set[0][1].flatten())
+            self.dim_output = training_snapshots[0].dim
         else:
-            self.parameters_dim = fom.parameters.dim
             self.dim_output = fom.dim_output
 
         self.__auto_init(locals())
```

Comment:
I am not sure about this little function. Should it stay although it is used in the super() function already?

Reply:
If you call it here again, you could remove the line with `self.validation_loss = validation_loss` and it would be safer when adding new arguments later. So I think I would keep it but remove the line above.